scripts/hash-exec-copy.py

```diff
@@ -0,0 +1,98 @@
+#! /usr/bin/env python3
+
+import argparse
+from datetime import datetime
+import requests
+import subprocess
+
+INPUT_DATE_FORMAT="%Y-%m-%d"
+REPO_URL = "/makerdao/community"
+
+def get_executive_by_title (exec_title):
+    base_git_api_url = f"https://api.github.com/repos{REPO_URL}/commits"
+
+    api_response = requests.get(
+        f"{base_git_api_url}?path=governance/votes/{exec_title}",
+        params = {'per_page': '1'}
+    )
+    api_response.raise_for_status()
+    commits = api_response.json()
+    if not commits:
+        raise ValueError(f"Executive copy not found: {exec_title}")
+
+    commit_hash = commits[0].get("sha", "")
+
+    # Get target exec copy content
+    exec_copy_response = requests.get(
+        f"https://raw.githubusercontent.com{REPO_URL}/{commit_hash}/governance/votes/{exec_title}"
+    )
+    exec_copy_response.raise_for_status()
+
+    # Get target exec copy content
+    executive_url = exec_copy_response.url
+
+    # Remove trailing whitespace as command substitution in shell script removes it and we want to keep the output consistent
+    # Example of shell script to create hash `cast keccak -- "$(wget 'PATH' -q -O - 2>/dev/null)"`
+    if exec_copy_response.text[-1] == '\n':
+        content = exec_copy_response.text[:-1]
+
+    return content, executive_url, commit_hash
+
+def get_content_hash (content):
+    try:
+        # Run the 'cast keccak' command with the content
+        keccak_result = subprocess.run(
+            ['cast', 'keccak', '--', content],  # Run the cast keccak command
+            stdout = subprocess.PIPE,  # Capture the output
+            stderr = subprocess.PIPE,  # Capture the errors
+            text = True,  # Return output as text, not bytes
+            check = True  # Raise an exception if the command fails
+        )
+        return keccak_result.stdout
+    except FileNotFoundError:
+        raise SystemExit("Error: The 'cast' command is not found. Ensure it's installed and in the PATH.")
+    except subprocess.CalledProcessError as e:
+        raise SystemExit(f"Command failed with error: {e.stderr}")
+
+def main():
+    # Parse positional arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("date", help=f"Date to find executive copy for (format: {INPUT_DATE_FORMAT})")
+    date_string = parser.parse_args().date.replace("date=", "")
+
+    try:
+        date = datetime.strptime(date_string, INPUT_DATE_FORMAT)
+    except ValueError:
+        raise SystemExit(f"Invalid date format. Please use {INPUT_DATE_FORMAT}.")
+
+    # Find the executive doc for the given date
+    POSSIBLE_EXEC_TITLES = [
+        f"Executive%20vote%20-%20{date.strftime('%B %d, %Y')}.md",
+        f"Executive%20Vote%20-%20{date.strftime('%B %d, %Y')}.md",
+        f"Executive%20vote%20-%20{date.strftime('%B %-d, %Y')}.md",
+        f"Executive%20Vote%20-%20{date.strftime('%B %-d, %Y')}.md"
```

Comment:
@SidestreamStrongStrawberry Could you please provide 4 command examples to test those 4 edge cases (as I assume you did it yourself)?

Reply:
Date | script command | bash command | Hashes match | Comment 
-- | -- | -- | -- | --
2024-03-06 | make exec-hash  date="2024-03-06" | cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/17c96c446ac3339c1f0b3081f6afffad26b7c508/governance/votes/Executive%20Vote%20-%20March%206,%202024.md' -q -O - 2>/dev/null)" |  ✅ | Exec doc title: [`Executive Vote - March 6, 2024.md`](https://github.com/makerdao/community/blob/master/governance/votes/Executive%20Vote%20-%20March%206%2C%202024.md)
2024-05-06 | make exec-hash  date="2024-05-06" | cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/be059e35221d04f91d7e2ed978175784e2f45607/governance/votes/Executive%20vote%20-%20May%2006,%202024.md' -q -O - 2>/dev/null)" | ✅  | Exec doc title: [`Executive vote - May 06, 2024.md`](https://github.com/makerdao/community/blob/master/governance/votes/Executive%20vote%20-%20May%2006%2C%202024.md)
2024-05-30 | make exec-hash  date="2024-05-30" | cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/16588abd76fefe0e1276ab630033facc541f954d/governance/votes/Executive%20vote%20-%20May%2030,%202024.md' -q -O - 2>/dev/null)" |  ✅ | Exec doc title: [`Executive vote - May 30, 2024.md`](https://github.com/makerdao/community/blob/master/governance/votes/Executive%20vote%20-%20May%2030%2C%202024.md)
2023-02-08 | make exec-hash  date="2023-02-08"  | cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/856ebe853b251537c4bfddb7daa92f9f2800260f/governance/votes/Executive%20vote%20-%20February%208,%202023.md' -q -O - 2>/dev/null)" |  ✅ | Exec doc title: [`Executive vote - February 8, 2023.md`](https://github.com/makerdao/community/blob/master/governance/votes/Executive%20vote%20-%20February%208%2C%202023.md)